flytekit/tools/fast_registration.py

```diff
@@ -226,12 +226,13 @@ def download_distribution(additional_distribution: str, destination: str):
     except FlyteDataNotFoundException as ex:
         raise RuntimeError("task execution code was not found") from ex
     tarfile_name = os.path.basename(additional_distribution)
-    if not tarfile_name.endswith(".tar.gz"):
+    if tarfile_name.endswith(".tar.gz"):
+        # This will overwrite the existing user flyte workflow code in the current working code dir.
+        result = subprocess.run(
+            ["tar", "-xvf", os.path.join(destination, tarfile_name), "-C", destination],
+            stdout=subprocess.PIPE,
+        )
+        result.check_returncode()
+    elif tarfile_name != "pkl.gz":
```

Comment:
Can this special `"pkl.gz"` value be stored somewhere?

Reply:
Do you mean that we should store `pkl.gz` as a global variable somewhere else?

Reply:
Yup. Looks like you already did it in https://github.com/flyteorg/flytekit/pull/2733/commits/1ed6960191ca5a34d2c2e7dd66081a574a40bcc5